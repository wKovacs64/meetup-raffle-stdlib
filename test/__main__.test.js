const {
  EVENT_ID_NOT_FOUND,
  EVENT_ID_NOT_PUBLIC,
  EVENT_ID_UNEXPECTED,
  MEETUP,
  MEETUP_NO_EVENTS,
  MEETUP_NOT_FOUND,
  WINNER_NAME,
} = require('../__mocks__/testData');
const winner = require('../functions/__main__');

jest.mock('meetup-randomizer', () =>
  // eslint-disable-next-line global-require
  require('../__mocks__/meetupRandomizer.mock')
);

describe('winner', () => {
  it('should reject if Meetup not found', () =>
    expect(winner(MEETUP_NOT_FOUND)).rejects.toHaveProperty(
      'message',
      expect.stringMatching(/^Sorry/)
    ));

  it('should reject if no upcoming Events found', () =>
    expect(winner(MEETUP_NO_EVENTS)).rejects.toHaveProperty(
      'message',
      expect.stringMatching(/^Sorry/)
    ));

  it('should reject if Event not found', () =>
    expect(winner(MEETUP, EVENT_ID_NOT_FOUND)).rejects.toHaveProperty(
      'message',
      expect.stringMatching(/^Sorry/)
    ));

  it('should reject if Event not public', () =>
    expect(winner(MEETUP, EVENT_ID_NOT_PUBLIC)).rejects.toHaveProperty(
      'message',
      expect.stringMatching(/^Sorry/)
    ));

  it('should reject on unexpected data', () =>
    expect(winner(MEETUP, EVENT_ID_UNEXPECTED)).rejects.toHaveProperty(
      'message',
      expect.stringMatching(/^Sorry/)
    ));

  it('should resolve with a winner for a valid Meetup Event', () =>
    expect(winner(MEETUP)).resolves.toMatchObject({ winners: [WINNER_NAME] }));
});
