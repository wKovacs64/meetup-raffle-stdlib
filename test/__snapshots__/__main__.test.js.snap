// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`winner should reject if Event not found 1`] = `"Sorry, I couldn't find any information on that."`;

exports[`winner should reject if Event not public 1`] = `"Sorry, their members list is private."`;

exports[`winner should reject if Meetup not found 1`] = `"Sorry, I couldn't find any information on that."`;

exports[`winner should reject if no upcoming Events found 1`] = `"Sorry, I couldn't find any upcoming events."`;

exports[`winner should reject on unexpected data 1`] = `"Sorry, we received unexpected data for that request."`;

exports[`winner should resolve with a winner for a valid Meetup Event 1`] = `
Object {
  "winners": Array [
    Object {
      "name": "Tiny Rick",
      "photoURL": "https://i.imgur.com/rgDv1wB.jpg",
      "profileURL": "http://rickandmorty.wikia.com/wiki/Tiny_Rick",
    },
  ],
}
`;
